documentation/user-consent.md

```diff
@@ -48,122 +43,269 @@ The document details aspects regarding the consent management, which includes fo
+>[TO BE EDITED/COMPLETED] - Consent capture by operator and storage in operator (consent master).
 
 ### Authorization flows / grant types
 
->[TO BE EDITED/COMPLETED] - Authorization flows / grant types to be considered... E.g.: Authorization code flow, JWT-bearer flow, CIBA flow, Client credentials flow. For each API it should be possible to categorize it depending on the use case, integration/GTM model, personal information processing requirements, etc... and define the grant type/s to be used for each API.
-
-#### Authorization code flow
+#### Authorization code flow (Frontend flow)
 
->[TO BE EDITED/COMPLETED] - Flow to obtain an access token to consume network API...
+```mermaid
+sequenceDiagram
+autonumber
+title Consume a CAMARA API - Authorization Code Grant (FrontEnd)
+participant FE as Device App
+participant BE as Invoker (App Backend/Aggregator)
+box Operator
+  participant ExpO as OGW Platform<br>@Operator
+  participant Consent as Consent Master<br>@Operator
+end
 
-#### JWT-bearer flow
+Note over FE,BE: Use Feature needing<br>Operator Capability  
+BE->>FE: Auth Needed - redirect <br>/authorize?response_type=code&client_id=coolApp<br>&purpose=purpose&redirect_uri=invoker_callback...
+FE->>+FE: Browser /<br> Embedded Browser
+alt Standard OIDC Auth Code Flow between Invoker and OGW Platform
+  FE-->>ExpO: GET /authorize?response_type=code&client_id=coolApp&purpose=purpose&redirect_uri=invoker_callback...
+  Note over ExpO: OGW Platform Applies<br>Network Based Authentication (amr=nba/mnba)
+  ExpO->>ExpO: Network Based Authentication:<br>- map to Telco Identifier e.g.: phone_number<br>- Set UserId (sub)  
+  ExpO->>ExpO: Check legal basis of the purpose<br> e.g.: contract, legitimate_interest, consent, etc 
+  opt If User Consent is required for the legal basis of the purpose  
+    ExpO->>Consent: Check if Consent is granted    
+  end  
+  alt If Consent is Granted or Consent not needed for legal basis   
+    ExpO-->>FE: 302<br>Location: invoker_callback?code=Operatorcode
+  else If Consent is NOT granted - Consent Capture within AuthCode Flow  
+    Note over FE,ExpO: Consent capture  
+    alt when user grants consent / otherwise flow fails  
+      ExpO-->>FE: 302<br>Location: invoker_callback?code=Operatorcode
+    end  
+  end  
+  FE-->>-BE: GET invoker_callback?code=OperatorCode
+  BE->>ExpO: POST /token<br> code=OperatorCode
+  ExpO->>BE: 200 OK <br> {OperatorAccessToken}
+end
+
+BE->>ExpO: Access Operator CAMARA API <br> Authorization: Bearer {OperatorAccessToken}        
+ExpO->>ExpO: Decrypt OperatorAccessToken,<br>grants Access,<br>progresses request to API Backend,<br>gets API response  
+ExpO->>BE: CAMARA API Response
+Note over BE,FE: Response
+```
+
+<br>
+
+**Flow description**:
+
+First, the API invoker (which could potentially be the application BE, an aggregator, etc.) instructs the application frontend in the device to initiate the OIDC authorization code flow with the operator. 
```

Comment:
BE abbreviation could be not clear:
```suggestion
First, the API invoker (which could potentially be the application Backend, an aggregator, etc.) instructs the application frontend in the device to initiate the OIDC authorization code flow with the operator. 
```

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/54/commits/c7f9715fa8c597e4c9f94e8f0bbf3aecc501ead5.